addons/im_livechat/models/discuss_channel.py

```diff
@@ -24,6 +24,7 @@ class DiscussChannel(models.Model):
     livechat_operator_id = fields.Many2one('res.partner', string='Operator')
     chatbot_current_step_id = fields.Many2one('chatbot.script.step', string='Chatbot Current Step')
     chatbot_message_ids = fields.One2many('chatbot.message', 'discuss_channel_id', string='Chatbot Messages')
+    allow_visitor_upload = fields.Boolean(compute="_compute_allow_visitor_upload", store=True)
```

Comment:
I would move it to the base code, there is value to prevent upload from guests/portal as well on some specific shared channels. This will also be less patches.